src/Components/Utilities/NavControls.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from 'react';
+import { IoIosTrendingUp } from 'react-icons/io';
+import { MdOutlineWatchLater, MdPlaylistAdd } from 'react-icons/md';
+import { VscHistory, VscHome } from 'react-icons/vsc';
+import { Link } from 'react-router-dom';
+import '../Styles/NavControls.css';
+
+const NavControls = () => {
+  const [activeControl, setActiveControl] = useState('home');
+
+  return (
+    <section className='controls flex-center'>
+      <Link to='/'>
+        <VscHome
+          className={`icon light-color ${
+            activeControl === 'home' ? 'active-control' : ''
+          }`}
+          onClick={(e) => setActiveControl('home')}
+        />
+      </Link>
+      <Link to='/trending'>
+        <IoIosTrendingUp
+          className={`icon light-color ${
+            activeControl === 'trending' ? 'active-control' : ''
+          }`}
+          onClick={(e) => setActiveControl('trending')}
+        />
+      </Link>
+      <Link to='/watchlater'>
+        <MdOutlineWatchLater
+          className={`icon light-color ${
+            activeControl === 'watchlater' ? 'active-control' : ''
+          }`}
+          onClick={(e) => setActiveControl('watchlater')}
+        />
+      </Link>
+      <Link to='/history'>
+        <VscHistory
+          className={`icon light-color ${
+            activeControl === 'history' ? 'active-control' : ''
+          }`}
+          onClick={(e) => setActiveControl('history')}
+        />
+      </Link>
+      <Link to='/playlists'>
+        <MdPlaylistAdd
+          className={`icon light-color ${
+            activeControl === 'playlist' ? 'active-control' : ''
+          }`}
+          onClick={(e) => setActiveControl('playlist')}
+        />
+      </Link>
```

Comment:
suggestion - jsx is mostly repeating here so can create an array of object of every nav item and loop over them using map

Reply:
sure